components/Chat/Chat.tsx

```diff
@@ -200,15 +200,18 @@ export const Chat = memo(({ stopConversationRef }: Props) => {
             }
           }
           saveConversation(updatedConversation);
+
+          let conversationExists = false;
           const updatedConversations: Conversation[] = conversations.map(
             (conversation) => {
               if (conversation.id === selectedConversation.id) {
+                conversationExists = true;
                 return updatedConversation;
               }
               return conversation;
             },
           );
-          if (updatedConversations.length === 0) {
+          if (!conversationExists) {
```

Comment:
Previous logic of `updatedConversations.length === 0` only added a new conversation if there wasn't any previous history. Co-opting this since now there is a possible flow where a new conversation can be added even if existing conversations exist (e.g. new page starting a new chat automatically).